profiles.wordpress.org/public_html/wp-content/plugins/wporg-5ftf-activity-handler/wporg-5ftf-activity-handler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/*
+Plugin Name: WordPress.org 5ftf Activity Handler
+Plugin URI: http://wordpress.org
+License: GPL2
+Version: 1.1
+Description: Handles saving of last 5ftf contribution.
+*/
+
+defined( 'ABSPATH' ) or die();
+
+if ( ! class_exists( 'WPOrg_5ftf_Activity_Handler' ) ) {
+
+	class WPOrg_5ftf_Activity_Handler {
+
+		/**
+		 *  Name of meta key that tracks last contribution as a unix timestamp.
+		 *
+		 * @var string
+		 */
+		const last_contribution_meta_key = 'wporg_5ftf_last_contribution';
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			add_filter( 'bp_activity_add', array( $this, 'handle_contribution' ) );
+		}
+
+		/**
+		 * Saves contribution if it qualifies as a contribution.
+		 */
+		public function handle_contribution( $args ) {
+			if ( self::is_5ftf_contribution( $args['type'] ) ) {
+				self::update_last_contribution_meta( $args['user_id'] );
+			}
+		}
+
+		/**
+		 * Returns whether action is considered a contribution.
+		 */
+		public function is_5ftf_contribution( $action ) {
+			$wordpress_actions = array( 'blog_post_create' );
+			$wordcamp_actions  = array( 'wordcamp_speaker_add', 'wordcamp_organizer_add' );
```

Comment:
Maybe it makes sense to keep different arrays for readability, maybe the prefix is enough,

Reply:
That's a good thought. Either way works for me 👍🏻